### PR TITLE
(PC-2340) implement componentWillUnmount in Discovery to delete tutos

### PR DESCRIPTION
**Pull request by Aliochka**

### CODE REVIEW

File: src/components/pages/discovery/__specs__/Discovery.spec.jsx [hunk 1]
```diff
@@ -53,4 +55,33 @@ describe('src | components | pages | discovery | Discovery', () => {
       expect(wrapper.state()).toStrictEqual(expected)
     })
   })
+
+  describe('at unmount', () => {
+    it('should delete tutos for new user', () => {
+      // given
+      props.tutoRecommendations = [{ id: 'hello' }]
```
Comment (NerOcrO) [inline src/components/pages/discovery/__specs__/Discovery.spec.jsx]: J'aurai préféré avoir l'information `productOrTutoIdentifier`.

File: src/components/pages/discovery/Discovery.jsx [hunk 2]
```diff
@@ -44,6 +44,19 @@ class Discovery extends PureComponent {
...
   }
 
+  componentWillUnmount() {
+    const { tutoRecommendations } = this.props
+    if (tutoRecommendations.length > 0) {
+      this.deleteTutoRecommendations(tutoRecommendations)
+    }
+  }
+
+  deleteTutoRecommendations = tutoRecommendations => {
```
[the remark this answers is not in the document]
Comment (Aliochka) [reply src/components/pages/discovery/Discovery.jsx]: :+1:

File: src/components/pages/discovery/DiscoveryContainer.js [hunk 3]
```diff
@@ -19,7 +20,7 @@ export const mapStateToProps = (state, ownProps) => {
   const { params } = match
   const { mediationId, offerId } = params
   const currentRecommendation = selectCurrentRecommendation(state, offerId, mediationId)
-
+  const tutoRecommendations = selectTutoRecommendations(state)
```
[the remark this answers is not in the document]
Comment (Aliochka) [reply src/components/pages/discovery/DiscoveryContainer.js]: :+1:

File: src/components/pages/discovery/DiscoveryContainer.js [hunk 4]
```diff
@@ -29,10 +30,15 @@ export const mapStateToProps = (state, ownProps) => {
     readRecommendations,
     recommendations,
     shouldReloadRecommendations,
+    tutoRecommendations,
   }
 }
 
 export const mapDispatchToProps = (dispatch, props) => ({
+  deleteRecommendation: recommendations => {
```
[the remark this answers is not in the document]
Comment (Aliochka) [reply src/components/pages/discovery/DiscoveryContainer.js]: :+1:

File: src/components/pages/discovery/DiscoveryContainer.js [hunk 5]
```diff
@@ -29,10 +30,15 @@ export const mapStateToProps = (state, ownProps) => {
...
+    dispatch(deleteData({ recommendations: recommendations }))
```
[the remark this answers is not in the document]
Comment (Aliochka) [reply src/components/pages/discovery/DiscoveryContainer.js]: :+1:

File: src/components/pages/discovery/__specs__/Discovery.spec.jsx [hunk 6]
```diff
@@ -53,4 +55,33 @@ describe('src | components | pages | discovery | Discovery', () => {
       expect(wrapper.state()).toStrictEqual(expected)
     })
   })
+
+  describe('at unmount', () => {
```
[the remark this answers is not in the document]
Comment (Aliochka) [reply src/components/pages/discovery/__specs__/Discovery.spec.jsx]: :+1:

File: src/components/pages/discovery/__specs__/Discovery.spec.jsx (hunk 1 above)
Comment (Aliochka) [reply src/components/pages/discovery/__specs__/Discovery.spec.jsx]: :+1:

File: src/components/pages/discovery/__specs__/DiscoveryContainer.spec.js [hunk 7]
```diff
@@ -171,6 +211,25 @@ describe('src | components | pages | discovery | DiscoveryContainer', () => {
           // then
           expect(ownProps.history.replace).toHaveBeenCalledWith('/decouverte/tuto/vide')
         })
+
+        it('should delete tuto recommendations when leaving discovery', () => {
```
[the remark this answers is not in the document]
Comment (Aliochka) [reply src/components/pages/discovery/__specs__/DiscoveryContainer.spec.js]: :+1:

File: src/components/pages/discovery/__specs__/DiscoveryContainer.spec.js [hunk 8]
```diff
@@ -171,6 +211,25 @@ describe('src | components | pages | discovery | DiscoveryContainer', () => {
...
+          // given
+          const tutoRecommendations = {
+            id: 'ABCD',
+          }
+          // when
```
[the remark this answers is not in the document]
Comment (Aliochka) [reply src/components/pages/discovery/__specs__/DiscoveryContainer.spec.js]: :+1:

File: src/components/pages/discovery/selectors/__specs__/selectTutoRecommendations.spec.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
+import selectTutoRecommendations from '../selectTutoRecommendations'
+
+describe('src | components | pages | discovery | selectors | selectTutoRecommendations', () => {
+  describe('when there are tutos in recommendation', () => {
+    it('should select the tutos recommendation', () => {
```
[the remark this answers is not in the document]
Comment (Aliochka) [reply src/components/pages/discovery/selectors/__specs__/selectTutoRecommendations.spec.js]: :+1:

**Pull request merged**